src/firefly/js/visualize/ui/CatalogSelectViewPanel.jsx

```diff
@@ -0,0 +1,623 @@
+                    let valC = inFields.catalog.value;
+                    const optList = getSubProjectOptions(catmaster, valP);
+                    if (action.payload.fieldKey === 'project') {
+                        valC = optList[0].value;//If project has changed, initiialise to the first subproject found
+                        inFields = updateMerge(inFields, 'catalog', {
+                            value: valC
+                        });
+                    }
+                    inFields = updateMerge(inFields, 'tableview', {
+                        selProj: valP,
+                        selCat: valC
+                    });
+
+                    // Reinit the table and catalog value:
+                    const catTable = getCatalogOptions(catmaster, valP, valC).option;
+
+                    let sizeFactor = 1;
+                    const method = get(inFields, 'spatial.value', SpatialMethod.Cone.value);
+                    if (method === SpatialMethod.Box.value) {
+                        sizeFactor = 2;
+                    }
+
+                    //No need to act on other fields if user change user wpt or spatial method
+                    if (action.payload.fieldKey === 'UserTargetWorldPt'
+                        || action.payload.fieldKey === 'conesize') {
+
+                        break;
+
+                    }
+                    let idx = get(inFields, 'cattable.indexClicked', 0);//Get current value
+                    if (action.payload.fieldKey === 'project'
+                        || action.payload.fieldKey === 'catalog') {
+                        idx = 0; // reset to first item of the catalog list
+                    }
+
+                    // User clicked on the table to select a catalog and needs to propagte the index in order
+                    // to get the item highlighted
+                    if (action.payload.fieldKey === 'cattable') {
+                        idx = catTable.findIndex((e) => {
+                            return e.value === action.payload.value;
+                        });
+
+                    }
+                    const radius = parseFloat(catTable[idx].cat[7]);
+
+                    inFields = updateMerge(inFields, 'cattable', {
+                        groupKey: gkey,
+                        fieldKey: 'cattable',
+                        indexClicked: idx,
+                        value: catTable[idx].value
+                    });
+                    inFields = updateMerge(inFields, 'conesize', {
+                        groupKey: gkey,
+                        fieldKey: 'conesize',
+                        max: sizeFactor * radius / 3600
+                    });
+                    break;
+                case FieldGroupCntlr.CHILD_GROUP_CHANGE:
+                    console.log('Child group change called...');
+                    break;
+                default:
+                    break;
+            }
+            return inFields;
+        }
```

Comment:
This is good! The code clearly gets the purpose of the reducer, that is reflected in the interaction between the fields.  It all appears to work.  It is a very complex panel. Congratulations! 

I do think you should try to break up the reducer in some several functions to make it easier to follow. No suggestion on how but I struggle with this for ColorPanelReducer. However, the hard part is done.